userservice/src/main/java/com/infodation/userservice/services/UserServiceImpl.java

```diff
@@ -21,6 +29,51 @@ public class UserServiceImpl implements IUserService {
     public UserServiceImpl(UserRepository userRepository) {
         this.userRepository = userRepository;
     }
+    @Async  // Annotation cho bất đồng bộ
+    public CompletableFuture<Void> importUsersFromCsv(MultipartFile file) throws IOException {
+        List<User> usersToSave = new ArrayList<>();
+
+        // Đọc file CSV
+        try (CSVReader reader = new CSVReader(new InputStreamReader(file.getInputStream()))) {
+            String[] nextLine;
+            boolean isHeader = true;
+
+            while ((nextLine = reader.readNext()) != null) {
+                if (isHeader) {  // Skip dòng tiêu đề
+                    isHeader = false;
+                    continue;
+                }
+                try {
+                    // Tạo đối tượng User, chỉ lấy các cột cần thiết
+                    User user = new User();
+                    user.setUserId(nextLine[1]); // Giả định cột 1 là userId
+                    user.setFirstName(nextLine[2]); // Giả định cột 2 là firstName
+                    user.setLastName(nextLine[3]); // Giả định cột 3 là lastName
+                    user.setEmail(nextLine[5]); // Giả định cột 5 là email
+                    user.setSex(Sex.valueOf(nextLine[4].toUpperCase())); // MALE/FEMALE
+                    user.setCreatedAt(LocalDateTime.now());
+                    user.setUpdatedAt(LocalDateTime.now());
+                    usersToSave.add(user);
+                    // Khi danh sách đủ 10.000 người dùng thì lưu
+                    if (usersToSave.size() == 5000) {
+                        userRepository.saveAll(usersToSave);
+                        usersToSave.clear(); // Xóa danh sách sau khi lưu
+                    }
+                } catch (ArrayIndexOutOfBoundsException | IllegalArgumentException e) {
+                    // Bỏ qua dòng có lỗi hoặc không đầy đủ
+                    System.out.println("Invalid data in CSV line, skipping this row.");
+                }
```

Comment:
When you try for particular exceptions, you also need to catch the general `Exception`.
```suggestion
                } catch (ArrayIndexOutOfBoundsException | IllegalArgumentException e) {
                    // Bỏ qua dòng có lỗi hoặc không đầy đủ
                    System.out.println("Invalid data in CSV line, skipping this row. Error: " + e.getMessage());
                }catch (Exception ex) {
                    // Bỏ qua dòng có lỗi hoặc không đầy đủ
                    System.out.println("Invalid data in CSV line, skipping this row. Error: " + e.getMessage());
                }
```